packages/shared/src/components/Feed.tsx

```diff
@@ -131,6 +131,7 @@ export default function Feed<T>({
     insaneMode,
     loadedSettings,
   } = useContext(SettingsContext);
+  const [isFetchingNextPage, setIsFetchingNextPage] = useState(false);
```

Comment:
You should use the state that is returned from `useFeed` for this purpose. This hook controls the requests for the feed. I think we already have an existing mechanism that prevents from fetching another page if there's a pending request. You can check inside the hook itself

Reply:
Yes, but we had to create a separate one since it might not know when the user actually clicked the next button. If the user just scrolled the background while the modal is open, it would trigger the `onNext`.